cmd/thv/app/secret.go

```diff
@@ -280,3 +293,132 @@ func getSecretsManager() (secrets.Provider, error) {
 
 	return manager, nil
 }
+
+func runSecretsSetup(cmd *cobra.Command, _ []string) error {
```

Comment:
Keep in mind that we will need similar logic in the API for the secret setup API. This implementation is probably ok for this PR, but you may need to split in future.